llpc/context/llpcCompiler.cpp

```diff
@@ -1460,7 +1489,13 @@ Result Compiler::buildGraphicsPipelineInternal(GraphicsContext *graphicsContext,
   if (buildingRelocatableElf) {
     result = buildPipelineWithRelocatableElf(context, shaderInfo, pipelineElf, stageCacheAccesses);
   } else {
+    const MetroHash::Hash originalPipelineHash = graphicsContext->getPipelineHashCodeWithoutCompact();
+    auto pipelineInfo = reinterpret_cast<const GraphicsPipelineBuildInfo *>(graphicsContext->getPipelineBuildInfo());
+    MetroHash::Hash cacheHash = {};
+    cacheHash = PipelineDumper::generateHashForGraphicsPipeline(pipelineInfo, true, false);
+    context->getPipelineContext()->setPipelineHashCode(convertToHashUsedForCacheLookup(cacheHash));
```

Comment:
I don't think you need to recompute and update the pipeline hash.  I should be set correctly coming into this function.  Look at around link 1585 where this function is called.  You will see that that graphics context initialized with the correct hash.  It needs to be changed for the relocatable shader compilation because the object being compiled is different than the entire pipeline.  We have changed it to be just a single shader.

Reply:
done